onchain/src/tests/art_peace.cairo

```diff
@@ -428,3 +428,75 @@ fn deposit_reward_test() {
     );
 }
 
+#[test]
+fn distribute_rewards_test() {
+    let art_peace_address = deploy_contract();
+    let art_peace = IArtPeaceDispatcher { contract_address: art_peace_address };
+    let template_verifier = ITemplateVerifierDispatcher { contract_address: art_peace.contract_address };
+    let template_store = ITemplateStoreDispatcher { contract_address: art_peace.contract_address };
+
+    let erc20_mock: ContractAddress = deploy_erc20_mock();
+    // let reward_amount: u256 = 1 * utils::pow_256(10, 18);
+    let reward_amount: u256 = 4;
+
+    let template_image = array![1, 2, 3, 4];
+    let template_hash = compute_template_hash(template_image.span());
+    let template_metadata = TemplateMetadata {
+        name: 'test',
+        hash: template_hash,
+        position: 0,
+        width: 2,
+        height: 2,
+        reward: reward_amount,
+        reward_token: erc20_mock,
+        creator: get_caller_address(),
+    };
+
+    let template_image_span = template_image.span();
+    let x = 10;
+    let y = 20;
+    let pos = x + y * WIDTH;
+    let color = 0x5;
+    let now = 10;
+    let template_id = 0;
+
+    let user = 123.try_into().unwrap();
+    let user2 = 1234.try_into().unwrap();
+    let user3 = 12345.try_into().unwrap();
+    let user4 = 123456.try_into().unwrap();
+
+    IERC20Dispatcher { contract_address: erc20_mock }.approve(art_peace_address, reward_amount);
+    let allowance = IERC20Dispatcher { contract_address: erc20_mock }.allowance(get_contract_address() ,art_peace_address);
+    println!("Allowance: {}", allowance);
+
+    template_store.add_template(template_metadata);
+    assert!(template_store.get_templates_count() == 1, "Templates count is not 1");
+
+    start_prank(CheatTarget::One(art_peace_address), user);
+    art_peace.place_pixel(pos, color, now);
+    stop_prank(CheatTarget::One(art_peace_address));
+
+    start_prank(CheatTarget::One(art_peace_address), user2);
+    art_peace.place_pixel(pos, color, now);
+    stop_prank(CheatTarget::One(art_peace_address));
+
+    start_prank(CheatTarget::One(art_peace_address), user3);
+    art_peace.place_pixel(pos, color, now);
+    stop_prank(CheatTarget::One(art_peace_address));
+
+    start_prank(CheatTarget::One(art_peace_address), user4);
+    art_peace.place_pixel(pos, color, now);
+    stop_prank(CheatTarget::One(art_peace_address));
```

Comment:
All users are setting the same pixel and the color is the wrong one.
You should use these values `(pos, color)`:
```
(0, 1)
(1, 2)
(WIDTH, 3)
(WIDTH + 1, 4)
```